### ISSUE TITLE
2023 DINOv2: Learning Robust Visual Features without Supervision

### ISSUE THREAD

**Issue by pomelyu** [pomelyu/paper-reading-notes#22]
## Introduction
This paper builds a general-purpose pretrained model that can be used in downstream tasks without fine-tuning. DINOv2 learned image features by discriminative self-supervised learning approach, and it doesn't need either labels or text description(eg. CLIP). This paper describes how they build curated dataset from a larger uncrated dataset for better image diversity, and also some improvements to reduce memory usage and speed up training for large dataset. In the downstream classification task, model with frozen pretrained layers and trainable linear classification layer outperform other state-of-the-art self-supervised model and achieve the performance of weakly self-supervised model, such as CLIP, but still worse than the best supervised model.

<img width="1292" alt="image" src="https://github.com/pomelyu/paper-reading-notes/assets/2679561/d1a5cc97-11ea-49bd-90d5-d60017a4db0f">

### Discriminative Self-supervised Pre-training
- Image-level objctive[^1]
- Patch-level objective[^2]
- Untyping head weights between both objectives
- Sinkhorn-Knopp centering[^3]
- KoLeo regularizer[^4]
- Adapting the resolution[^5]

### Efficient implementation
- Fast and memory-efficient attention: slightly modify the dimension used in Vit structure to fit hardware specification.
- Sequence packing: DINO algorithm use two different cropping sizes and thus generate embeddings with different length. The paper used sequence packing derived from NLP. Specifically, they concatenate these two sequences into a long sequence and pass it through the transformer and use attention mask to prevent the attention between different sequences.
- Efficient stochastic depth: skip the computation of dropped residuals instead of masking the result
- Fully-Shared Data Parallel(FSDP): split the model across GPUs by using the pytorch implementation of FSDP
- Model distillation: The paper said they can use their training loop to perform distillation instead of training a distilled model from scratch

## Highlight
- method to stabilize training for larger dataset [^1]
- method to accelerate training and reduce RAM usage
- method to curate large dataset base on a small one
- method for model distillation
- larger model perform much better in large dataset, but has the same performance for small dataset
![image](https://github.com/pomelyu/paper-reading-notes/assets/2679561/4be05dda-27ea-4e60-b5bc-fa671284a866)

- In their case, training on small resolution and then fine-tuning on high resolution get similar results from training on high resolution from scratch.
- PCA of patch features show the parts of objects match well for images of the same category
![image](https://github.com/pomelyu/paper-reading-notes/assets/2679561/f01f4330-3434-46af-8b4d-961928ec92de)

## Comments

[^1]: Emerging properties in self-supervised vision transformers. In ICCV, 2021.
[^2]: ibot: Image bert pre-training with online tokenizer. In ICLR, 2022a.
[^3]: Unsupervised learning of visual features by contrasting cluster assignments. In NeurIPS, 2020
[^4]: Spreading vectors for similarity search. In ICLR, 2019.
[^5]: Fixing the train-test resolution discrep- ancy. In NeurIPS, 2019.
